src/main/java/bio/terra/testrunner/runner/TestRunner.java

```diff
@@ -154,6 +156,28 @@ private void executeTestConfigurationNoGuaranteedCleanup() throws Exception {
       logger.info("Version: Skipping version determination");
     }
 
+    if (config.server.testRunnerEnvironmentScript != null) {
+      TestRunnerEnvironmentScript ghContextScript =
+          config
+              .server
+              .testRunnerEnvironmentScript
+              .scriptClass
+              .getDeclaredConstructor()
+              .newInstance();
+      gitHubContextResult = ghContextScript.getTestRunnerEnvironmentContext();
+    } else {
+      try {
+        Class<?> scriptClassGeneric = GitHubActionsWorkflowRunContext.class;
+        Class<? extends TestRunnerEnvironmentScript> scriptClass =
+            (Class<? extends TestRunnerEnvironmentScript>) scriptClassGeneric;
+        gitHubContextResult =
```

Comment:
What happens when you run TestRunner locally? Does this cause errors, or just populate a `TestRunnerEnvironmentScriptResult` with empty values?

Reply:
The idea is that `githubRunId, githubRepository, and githubServerUrl` will always be captured from the corresponding envvars when available. If no such envvars are available, then they will just be `null` values. Yes the `GitHubActionsWorkflowRunContext` will always be run so these values can be used by `trdash` to display results grouped by the run id.